Add support for NetBSD

bkuzma-nowaera opened the pull request.

Commit:
Add support for NetBSD

DrHyde commented:
While the code in this PR _builds_ on NetBSD it gives incorrect results:

```
$ ./duf --only local --style ascii
+-------------------------------------------------------------------------------------+
| 1 local device                                                                      |
+------------+-------+-------+-------+----------------------------+------+------------+
| MOUNTED ON |  SIZE |  USED | AVAIL |            USE%            | TYPE | FILESYSTEM |
+------------+-------+-------+-------+----------------------------+------+------------+
| /          | 58.2G | 24.4G | 30.8G | [########..........] 42.0% | ffs  | /dev/dk0   |
+------------+-------+-------+-------+----------------------------+------+------------+

$ df -h /
Filesystem     Size   Used  Avail %Cap Mounted on
/dev/dk0       7.3G   3.1G   3.9G  44% /
```

This is with NetBSD 10.1 on i386 (32-bit, not 64-bit), built using Go 1.25.2.

I find it deeply suspicious that the size/used/avail figures are all off by approximately a factor of 8 - and the "approximately" could easily be a result of rounding. Is the underlying block size not being retrieved correctly?

Commit:
Fix incorrect sizes reported by duf.

According to sys/statvfs.h blocks, bfree, bavail are in units of frsize not bsize.

bkuzma-nowaera commented:
According to sys/statvfs.h blocks, bfree, bavail are in units of frsize not bsize.

DrHyde commented:
Thanks, with that patch I get the right results
